lib/mini_scheduler/manager.rb

```diff
@@ -352,8 +384,11 @@ def self.seq
       end
     end
 
+    @@identity_key_mutex = Mutex.new
     def identity_key
-      @identity_key ||= "_scheduler_#{hostname}:#{Process.pid}:#{self.class.seq}:#{SecureRandom.hex}"
+      @@identity_key_mutex.synchronize do
```

Comment:
do you need to acquire the mutex if @identity_key is already defined? Just shortcut and return it?

Reply:
Updated